src/main/scala/dpla/ingestion3/EnrichEntry.scala

```diff
@@ -103,23 +110,31 @@ object EnrichEntry {
       .format("com.databricks.spark.avro")
       .save(outputDir)
 
-    // Gather some stats
-    val mappedRecordCount = mappedRows.count()
-    val enrichedRecordCount = successResults.count()
-
     sc.stop()
 
     // Log error messages.
     failures.foreach(msg => enrichLogger.warn(s"Error: ${msg}"))
 
-    enrichLogger.debug(s"Mapped ${mappedRecordCount} records and enriched ${enrichedRecordCount} records")
-    enrichLogger.debug(s"${mappedRecordCount-enrichedRecordCount} enrichment errors")
+    enrichLogger.debug(
+      s"Mapped ${totalCount.value} records and enriched ${successCount.value}" +
+        s" records"
+    )
+    enrichLogger.debug(
+      s"${totalCount.value - successCount.value} enrichment errors"
+    )
   }
 
-  private def enrich(dplaMapData: OreAggregation, driver: EnrichmentDriver): (Row, String) = {
+  private def enrich(dplaMapData: OreAggregation,
+                     driver: EnrichmentDriver,
+                     totalCount: LongAccumulator,
+                     successCount: LongAccumulator): (Row, String) = {
+    totalCount.add(1)
     driver.enrich(dplaMapData) match {
-      case Success(enriched) => (RowConverter.toRow(enriched, model.sparkSchema), null)
-      case Failure(exception) => (null, exception.getMessage)
+      case Success(enriched) =>
+        successCount.add(1)
+        (RowConverter.toRow(enriched, model.sparkSchema), null)
+      case Failure(exception) =>
+        (null, exception.getMessage)
```

Comment:
There's a bit in the story about changing this so the whole exception stack trace gets turned into a string rather than just the message.

Reply:
Ah looks like you did it in MappingEntry. Do it here as well?